File: backend/src/main/java/com/java/NBE4_5_1_7/domain/study/controller/StudyMemoLikeController.java
package com.java.NBE4_5_1_7.domain.study.controller;

import com.java.NBE4_5_1_7.domain.study.service.StudyMemoLikeService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("api/v1/studyMemo/like")
@RequiredArgsConstructor
public class StudyMemoLikeController {
    private final StudyMemoLikeService studyMemoLikeService;

    @PostMapping("/{studyMemoId}")
    public ResponseEntity<String> saveStudyMemoLike(@PathVariable Long studyMemoId) {
        return ResponseEntity.ok(studyMemoLikeService.memoLike(studyMemoId));
    }
}

File: backend/src/main/java/com/java/NBE4_5_3_7/domain/study/controller/StudyMemoLikeController.kt
package com.java.NBE4_5_3_7.domain.study.controller

import com.java.NBE4_5_3_7.domain.member.service.MemberService
import com.java.NBE4_5_3_7.domain.study.service.StudyMemoLikeService
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("api/v1/studyMemo/like")
class StudyMemoLikeController(
    private val studyMemoLikeService: StudyMemoLikeService,
    private val memberService: MemberService
) {
    @GetMapping("/{studyMemoId}/status")
    fun getLikeStatus(@PathVariable studyMemoId: Long): ResponseEntity<Map<String, Any>> {
        val member = memberService.getMemberFromRq()
        return ResponseEntity.ok(studyMemoLikeService.getLikeStatus(studyMemoId, member))
    }

    @PostMapping("/{studyMemoId}")
    fun saveStudyMemoLike(@PathVariable studyMemoId: Long): ResponseEntity<String> {
        val member = memberService.getMemberFromRq()
        return ResponseEntity.ok(studyMemoLikeService.memoLike(studyMemoId, member))
    }
}
